File: 2112-minimum-difference-between-highest-and-lowest-of-k-scores/minimum-difference-between-highest-and-lowest-of-k-scores.go
func minimumDifference(nums []int, k int) int {
    if len(nums) < k || len(nums) == 1 {
        return 0
    }
    sort.Ints(nums);
    min := float64(100000)
    gap := k - 1;
    for i := len(nums) - 1; i >= gap; i-- {
        min = math.Min(float64(min), float64(nums[i] - nums[i - gap]))
    } 
    return int(min);
}

File: 2112-minimum-difference-between-highest-and-lowest-of-k-scores/minimum-difference-between-highest-and-lowest-of-k-scores.ts
function minimumDifference(nums: number[], k: number): number {
    nums.sort((a, b)=> a - b);
    const end = nums.length - k + 1;
    let diff = Number.MAX_SAFE_INTEGER;
    for(let i = 0; i < end; i++) {
        diff = Math.min(diff, nums[i + k - 1] - nums[i]);
    }
    return diff;
};